dotcom-rendering/src/lib/byline.ts

```diff
@@ -1,11 +1,38 @@
-export const getContributorTags = (tags: TagType[]): TagType[] => {
-	return tags.filter((t) => t.type === 'Contributor');
+interface ContributorTag extends TagType {
+	type: 'Contributor';
+}
+
+const isContributor = (tag: TagType): tag is ContributorTag =>
+	tag.type === 'Contributor';
+
+/**
+ * Get the sole contributor of an article.
+ * This only works if all of the following are true:
+ * - there is only one `Contributor` tag
+ * - that tag’s `title` is the same as the article byline
+ *
+ * @returns A sole contributor if there is one, undefined otherwise.
+ */
+export const getSoleContributor = (
+	tags: TagType[],
+	byline: string | undefined,
+): ContributorTag | undefined => {
+	const [soleContributor, ...otherContributors] = tags.filter(isContributor);
```

Comment:
if I understand the code correctly, then if the right hand side is an empty array, `soleContributor` will be `undefined` and `otherContributors` will be `[]`, right? This seems like a very neat way to write this bit of logic! And I think the naming of the variables makes it easy to get a sense of what's going on. But I wonder whether the underlying mechanisms might be a bit opaque to future readers? (I know that I had to double check what using rest decomposition on an empty array would yield, at least.)

Reply:
I'll add a comment about what destructuring does on an empty array.

If we enabled `strictNullChecks` in our `.tsconfig`, the compiler would know that soleContributor could be `undefined`.